### PR TITLE
fix: auto expose ports for existing Deployment*

### LINKED ISSUE #1356: Auto exposure of containerPorts only work for swapDeployment
##### Issue Overview

Telepresence support to auto expose the container ports defined in the Deployment/DeploymentConfig. This behavior would be useful regardless of the chosen deployment handling strategy.

##### Expected Behaviour

Should work with  `--new-deployment`, `--deployment` and `--swap-deployment`

##### Current Behaviour

Only `--swap-deployment` is currently supported.

### PR DESCRIPTION
**Pull request by aslakknutsen**
fixes #1356

### CODE REVIEW

File: telepresence/proxy/deployment.py [hunk 1]
```diff
@@ -113,6 +125,19 @@ def existing_deployment_openshift(
...
+    deployment, container = _split_deployment_container(deployment_arg)
+    container_to_update = _get_container_name(container, d_json)
+
+    for container in d_json["spec"]["template"]["spec"]["containers"]:
+        if container["name"] == container_to_update:
+            # Merge container ports into the expose list
+            expose.merge_automatic_ports([
+                port["containerPort"] for port in container.get("ports", [])
+                if port["protocol"] == "TCP"
+            ])
```
Comment (marun) [inline telepresence/proxy/deployment.py:138]: nit: Maybe exit the loop once the container has been found?

Comment (aslakknutsen) [reply telepresence/proxy/deployment.py:138]: Fixed https://github.com/telepresenceio/telepresence/pull/1358/commits/277c8eb89d66864119e61cc7fb774445eda37611

**Comment by aslakknutsen** [comment]
@marun Thank you for the review. PR should be up to date now. :)

**Comment by bartoszmajsak** [comment]
LOL sorry for approving that - I was looking at PR related to this one and somehow got distracted and came back to the wrong GH tab :)

**Comment by marun** [comment]
@aslakknutsen Thank you for the updates, LGTM

**Pull request merged**